### ISSUE TITLE
Multipart upload from Publisher<ByteBuf>

### ISSUE THREAD

**Issue by Rattenkrieg** [reactor/reactor-netty#2418]
I can't find a way to encode async sequence of `ByteBuf`  into `HttpClientForm`. There are few methods encoding `java.io.HttpClientForm`, `java.io.File` and plain byte arrays, however [HttpClientForm](https://github.com/reactor/reactor-netty/blob/main/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientForm.java) lacking something like 
```java
HttpClientForm file(String name, Publisher<? extends ByteBuf> partPublisher, @Nullable String contentType)
```
Are there any fundamental blockers for such api or is it just a lack of demand? In a latter case I could try to code a missing bits if core team is not working on it already.

**Comment by Rattenkrieg** [comment]
@violetagg I'm currently on 1.0.21, but it won't be an issue for me to migrate to 2.0.0. However there is pending #2327 which I guess is a blocker for any multipart features in a 2.0.x branch. So probably 1.0.x is a better place to work until #2327 is merged.

**Comment by violetagg** [comment]
> @violetagg I'm currently on 1.0.21, but it won't be an issue for me to migrate to 2.0.0. However there is pending #2327 which I guess is a blocker for any multipart features in a 2.0.x branch. So probably 1.0.x is a better place to work until #2327 is merged.

Yep that's true, multipart is currently in progress for Reactor Netty 2.0.0

Agree, please base your work on `1.0.x` branch.

**Comment by violetagg** [comment]
Let me close that. There aren't plans for supporting this for the time being.
